pkg/controller/datadogagent/utils.go

```diff
@@ -771,6 +741,52 @@ func getVolumesForAgent(dda *datadoghqv1alpha1.DatadogAgent) []corev1.Volume {
 	return volumes
 }
 
+// getVolumesForConfig returns the volumes necessary to configure the Agent.
+func getVolumesForConfig(dda *datadoghqv1alpha1.DatadogAgent) []corev1.Volume {
+	confdVolumeSource := corev1.VolumeSource{
+		EmptyDir: &corev1.EmptyDirVolumeSource{},
+	}
+	if dda.Spec.Agent.Config.Confd != nil {
+		confdVolumeSource = corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: dda.Spec.Agent.Config.Confd.ConfigMapName,
+				},
+			},
+		}
+	}
```

Comment:
You are using `getVolumeForConfig()` for the `Agent` and the `ClusterCheckRunner`, but for the  "runner" we don't need to mount the "Confd" `confimap` since all the check configs will come from the `cluster-agent`